From b43d6d42d734e78bd9b805a62f2a64933c144a41 Mon Sep 17 00:00:00 2001
From: Sumit Garg <sumit.garg@nxp.com>
Date: Thu, 14 Jul 2016 07:20:02 -0400
Subject: [PATCH 80/93] ls1012ardb: PPA: Enable PPA validation in case of
 secure boot

As part of Secure Boot Chain of trust, enable validation of PPA image
before the image is started.

Signed-off-by: Sumit Garg <sumit.garg@nxp.com>
---
 arch/arm/include/asm/fsl_secure_boot.h |    2 ++
 include/configs/ls1012a_common.h       |    2 --
 include/configs/ls1012ardb.h           |    2 ++
 3 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/include/asm/fsl_secure_boot.h b/arch/arm/include/asm/fsl_secure_boot.h
index b2e9105..0580fcd 100644
--- a/arch/arm/include/asm/fsl_secure_boot.h
+++ b/arch/arm/include/asm/fsl_secure_boot.h
@@ -107,6 +107,8 @@
 #ifdef CONFIG_SYS_LS_PPA_FW_IN_XIP
 #ifdef CONFIG_LS1043A
 #define CONFIG_SYS_LS_PPA_ESBC_ADDR	0x600c0000
+#else
+#define CONFIG_SYS_LS_PPA_ESBC_ADDR	0x40480000
 #endif
 #else
 #error "No CONFIG_SYS_LS_PPA_FW_IN_xxx defined"
diff --git a/include/configs/ls1012a_common.h b/include/configs/ls1012a_common.h
index 0d64889..7e935ad 100644
--- a/include/configs/ls1012a_common.h
+++ b/include/configs/ls1012a_common.h
@@ -191,6 +191,4 @@
 #define CONFIG_DOS_PARTITION
 #define CONFIG_CMD_EXT2
 
-#include <asm/fsl_secure_boot.h>
-
 #endif /* __LS1012A_COMMON_H */
diff --git a/include/configs/ls1012ardb.h b/include/configs/ls1012ardb.h
index e97c1f9..d78edd2 100644
--- a/include/configs/ls1012ardb.h
+++ b/include/configs/ls1012ardb.h
@@ -149,4 +149,6 @@
 
 #define CONFIG_FSL_CAAM			/* Enable CAAM */
 
+#include <asm/fsl_secure_boot.h>
+
 #endif /* __LS1012ARDB_H__ */
-- 
1.7.9.5

